roles/jd-client/src/lib/mod.rs

```diff
@@ -119,53 +119,58 @@ impl JobDeclaratorClient {
-                        task_collector
-                            .safe_lock(|s| {
-                                for handle in s {
-                                    handle.abort();
-                                }
-                            })
-                            .unwrap();
-                        upstream_index += 1;
-                        tokio::time::sleep(std::time::Duration::from_secs(1)).await;
-                        break;
-                    }
-                    status::State::Healthy(msg) => {
-                        info!("HEALTHY message: {}", msg);
+                // let task_status: status::Status = task_status.unwrap();
+
+                if let Ok(task_status) = task_status {
+                    match task_status.state {
+                        // Should only be sent by the downstream listener
+                        status::State::DownstreamShutdown(err) => {
+                            error!("SHUTDOWN from: {}", err);
+                            tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+                            task_collector
+                                .safe_lock(|s| {
+                                    for handle in s {
+                                        handle.abort();
+                                    }
+                                })
+                                .unwrap();
+                            tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+                            break;
+                        }
+                        status::State::UpstreamShutdown(err) => {
+                            error!("SHUTDOWN from: {}", err);
+                            tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+                            task_collector
+                                .safe_lock(|s| {
+                                    for handle in s {
+                                        handle.abort();
+                                    }
+                                })
+                                .unwrap();
+                            tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+                            break;
+                        }
+                        status::State::UpstreamRogue => {
+                            error!("Changin Pool");
+                            tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+                            task_collector
+                                .safe_lock(|s| {
+                                    for handle in s {
+                                        handle.abort();
+                                    }
+                                })
+                                .unwrap();
+                            upstream_index += 1;
+                            tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+                            break;
```

Comment:
would this `break` and the other ones not specifying the loop tag, break the inner or outer one?